integration/noderunner/noderunner_test.go

```diff
@@ -60,9 +60,12 @@ func TestCanStartStandaloneObscuroHostAndEnclave(t *testing.T) {
 		t.Fatal(err)
 	}
 	network := gethnetwork.NewGethNetwork(int(gethPort), int(gethWebsocketPort), gethBinaryPath, 1, 1, []string{address.String()})
-	defer network.StopNodes()
 	go enclaverunner.RunEnclave(enclaveConfig)
 	go hostrunner.RunHost(hostConfig)
+	obscuroClient := obscuroclient.NewClient(clientServerAddr)
+
+	defer network.StopNodes()
+	defer obscuroClient.Call(nil, obscuroclient.RPCStopHost) //nolint:errcheck
```

Comment:
I wonder if this is an example of where we could get into trouble between tests.

That call will return immediately now and then we plow onwards but if teardown took some time then shenanigans could ensue with the port not getting freed straight away. Worth waiting for the connection to die before exiting the defer or am I being paranoid?

Reply:
The problem I found was that you can't wait for the stop call to finish, since the stop call shuts down the RPC server itself, preventing a response from ever coming back.

I could wait until the port is released in a loop, if you think that's a good idea?

Reply:
Yeah it's not elegant but I think it's worth doing just to give us peace of mind that it's not a possibility.